### ISSUE TITLE
Editor is running workers on UI thread

### ISSUE THREAD

**Issue by ecraig12345** [microsoft/fluentui#11489]
The live code editor on the website is supposed to use web workers on separate threads for TypeScript and other purposes, to prevent this resource-intensive code from running on the UI thread and causing lag. However, there's a built-in browser security restriction that code from a different origin/domain than the original page (such as a CDN) can't create workers ([details here, sort of](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers)). It's supposed to be possible to work around this by using blob URLs or data URIs to load the code, and we tried to do this, but at least in Chrome it's currently falling back to running workers on the main thread. ~~Suspect this has to do with changing `configureEnvironment` to use blob URLs instead of data URIs.~~

~~We may need to vary the approach based on what works per-browser--data URIs seemed to work in Chrome but not old Edge, and blobs may work in old Edge but not Chrome.~~

**EDIT:** it still runs on the UI thread in Chrome/new Edge with data URIs, at least per the console warnings. 😞

**Comment by ecraig12345** [comment]
Still not totally fixed. :(

**Comment by ecraig12345** [comment]
On second thought, I'm re-closing this...

Even if the workers are running on the UI thread, that's not ideal, but the real question is whether it's causing any problems in practice. From my observation, the answer is not really: granted I haven't used the example editor extensively, and I have a fast computer, but I haven't noticed a disruptive amount of lag in my interactions with the editor. So this is probably not a super high priority to fix.

The other confusing part is that the [monaco repo](https://github.com/microsoft/monaco-editor) used to have documentation about the cross-domain scenario, but all of that has been removed...so it's even less clear what (if anything) can or should be done to resolve this. 

It might be worth trying `monaco-editor-webpack-plugin` again to see if that helps--back when we were originally working on the editor, we stopped using the plugin because we couldn't get it to do what we wanted (and the internals were...opaque), but maybe there have been improvements since then.
